supervisor/core.py

```diff
@@ -355,27 +355,24 @@ async def _adjust_system_datetime(self):
         except WhoamiError as err:
             _LOGGER.warning("Can't adjust Time/Date settings: %s", err)
             return
-        else:
-            if not self.sys_config.timezone:
-                self.sys_config.timezone = data.timezone
-            return
 
-        # Adjust timesettings in case SSL fails
-        try:
-            data = await retrieve_whoami(self.sys_websession, with_ssl=False)
-        except WhoamiError as err:
-            _LOGGER.error("Can't adjust Time/Date settings: %s", err)
-            return
-        else:
-            if not self.sys_config.timezone:
-                self.sys_config.timezone = data.timezone
+        # SSL Date Issue & possible time drift
+        if not data:
+            try:
+                data = await retrieve_whoami(self.sys_websession, with_ssl=False)
+            except WhoamiError as err:
+                _LOGGER.error("Can't adjust Time/Date settings: %s", err)
+                return
```

Comment:
This could be moved inside `except WhoamiSSLError:` as that is the only situation where this would apply.